File: examples/src/main/java/completion/CompletionExample.java
package completion;

import com.cjcrafter.openai.OpenAI;
import com.cjcrafter.openai.completions.CompletionRequest;
import io.github.cdimascio.dotenv.Dotenv;

/**
 * In this Java example, we will be using the Legacy Completion API to generate
 * a response.
 */
public class CompletionExample {

    public static void main(String[] args) {

        // To use dotenv, you need to add the "io.github.cdimascio:dotenv-kotlin:version"
        // dependency. Then you can add a .env file in your project directory.
        String key = Dotenv.load().get("OPENAI_TOKEN");
        OpenAI openai = OpenAI.builder()
                .apiKey(key)
                .build();

        String input = "What is 9+10?";

        // We use 128 tokens, so we have minimal delay before the response (for testing).
        CompletionRequest request = CompletionRequest.builder()
                .model("davinci")
                .prompt(input)
                .maxTokens(128).build();

        System.out.println("Generating Response...");
        String text = openai.createCompletion(request).get(0).getText();
        System.out.println(text);
    }
}

File: examples/src/main/kotlin/completion/CompletionExample.kt
package completion

import com.cjcrafter.openai.completions.completionRequest
import com.cjcrafter.openai.openAI
import io.github.cdimascio.dotenv.dotenv

/**
 * In this Kotlin example, we will be using the Completions API to generate a response.
 */
fun main() {

    // To use dotenv, you need to add the "io.github.cdimascio:dotenv-kotlin:version"
    // dependency. Then you can add a .env file in your project directory.
    val key = dotenv()["OPENAI_TOKEN"]
    val openai = openAI { apiKey(key) }

    // Here you can change the model's settings, add tools, and more.
    val request = completionRequest {
        model("davinci")
        prompt("The wheels on the bus go")
    }

    val completion = openai.createCompletion(request)[0]
    println(completion.text)
}
